pkg/util/hack/type_checksum.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hack
+
+import (
+	"encoding/binary"
+	"hash"
+	"hash/crc64"
+	"reflect"
+)
+
+// CalcTypeChecksum return a value which can represent the change of the type T
+// It's used in the test to make sure the developer can read some comments or
+// make some related changes when the type T is changed.
+func CalcTypeChecksum[T any]() uint64 {
```

Comment:
It seems that the result of this function depends on the go version :thinking:. I don't know why, every steps look consistent. Let's hold this PR for a while.